go/controller/internal/controller/common.go

```diff
@@ -309,3 +308,9 @@ func defaultErrMessage(err error, defaultMessage string) string {
 
 	return defaultMessage
 }
+
+const notFoundOrReadyError = "unable to find referenced object or is not ready yet" // TODO
+
+func notFoundOrReadyErrorMessage(err error) string {
+	return fmt.Sprintf("unable to find referenced object or is not ready yet, got error: %s", err.Error())
```

Comment:
maybe change this format string to:

"Referenced object is either not found or not ready, found error: %s"